### PR TITLE
updated deps for routing-controllers 0.10.0

### PR DESCRIPTION
**Pull request by leonardlin**
updated peer dependency for routing-controllers 0.10.0

See also: https://github.com/epiphone/routing-controllers-openapi/issues/120

### CODE REVIEW

**Comment by leonardlin** [comment]
@epiphone it seems that [routing-controllers](https://github.com/typestack/routing-controllers) has awaken again and has made urgent dependency updates.

I created a PR to go along the newest version, would you mind reviewing the PR?

**Comment by leonardlin** [comment]
Hi @epiphone could you take a look at this PR? routing-controller has to come to life again and this PR is an update to match the new routing-controller version

File: src/generateSpec.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export function getPathParams(route: IRoute): oa.ParameterObject[] {
       const param: oa.ParameterObject = {
         in: 'path',
         name,
-        required: !token.optional,
+        required: !(token.modifier == '?'),
```
Comment (epiphone) [inline src/generateSpec.ts:133]:
```suggestion
        required: token.modifier !== '?',
```

File: __tests__/fixtures/spec.json [hunk 2]
```diff
@@ -387,6 +387,7 @@
             "name": "from",
             "required": true,
             "schema": {
+              "pattern": "[^\\/#\\?]+?",
```
Comment (epiphone) [inline __tests__/fixtures/spec.json:390]: What's up with this change?

**Review by epiphone — CHANGES_REQUESTED** [review]
Thank you for the contribution! I enabled the CI pipeline here, please make sure all the tests pass.

File: __tests__/fixtures/spec.json (hunk 2 above)
Comment (leonardlin) [reply __tests__/fixtures/spec.json:390]: This was because of updates in "path-to-regexp": "^6.2.1",

File: src/generateSpec.ts (hunk 1 above)
Comment (leonardlin) [reply src/generateSpec.ts]: fixed

**Comment by leonardlin** [comment]
@epiphone hope the updated PR is ok now.
linting and tests pass locally.

**Review by epiphone — APPROVED** [review]

**Comment by epiphone** [comment]
Published in https://github.com/epiphone/routing-controllers-openapi/releases/tag/v4.0.0

**Pull request merged**